Client/UserResearch/OnboardingUserResearch.swift

```diff
@@ -42,23 +45,62 @@ class OnboardingUserResearch {
     }
     
     // MARK: Initializer
-    init(lpVariable: LPVar? = LPVariables.showOnboardingScreen) {
+    init(lpVariable: LPVar? = LPVariables.showOnboardingScreenAB) {
         self.lpVariable = lpVariable
     }
     
     // MARK: public
     func lpVariableObserver() {
-        Leanplum.onVariablesChanged {
-            self.updatedLPVariables?(self.lpVariable)
+        // Condition: Leanplum is disabled
+        // If leanplum is not enabled then we set the value of onboarding research to true
+        // True = .variant 1 which is our default Intro View
+        // False = .variant 2 which is our new Intro View that we are A/B testing against
+        // and get that from the server
+        guard LeanPlumClient.shared.getSettings() != nil else {
+            self.updateValue(onboardingScreenType: .versionV1)
+            self.updatedLPVariable?()
+            return
+        }
+        // Condition: Update from leanplum server
+        // Get the A/B test variant from leanplum server
+        // and update onboarding user reasearch
+        LeanPlumClient.shared.finishedStartingLeanplum = {
+            let showScreenA = LPVariables.showOnboardingScreenAB?.boolValue()
+            LeanPlumClient.shared.finishedStartingLeanplum = nil
+            self.updateTelemetry()
+            let screenType: OnboardingScreenType = (showScreenA ?? true) ? .versionV1 : .versionV2
+            self.updateValue(onboardingScreenType: screenType)
+            self.updatedLPVariable?()
+        }
+        // Conditon: Leanplum server too slow
+        // We don't want our users to be stuck on Onboarding
+        // Wait 2 second and update the onboarding research variable
+        // with true (True = .variant 1)
+        // Ex. Internet connection is unstable due to which
+        // leanplum isn't loading or taking too much time
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+            guard LeanPlumClient.shared.finishedStartingLeanplum != nil else {
+                return
+            }
+            let lpStartStatus = LeanPlumClient.shared.startCallFinished
+            var lpVariableValue: OnboardingScreenType = .versionV1
+            // Condition: LP has already started but we missed onStartLPVariable callback
+            if lpStartStatus, let boolValue = LPVariables.showOnboardingScreenAB?.boolValue() {
+                lpVariableValue = boolValue ? .versionV1 : .versionV2
+                self.updateTelemetry()
+            }
+            self.updatedLPVariable = nil
+            self.updateValue(onboardingScreenType: lpVariableValue)
+            self.updatedLPVariable?()
         }
     }
     
-    func updateValue(value: Bool) {
+    func updateValue(onboardingScreenType: OnboardingScreenType) {
         // For LP variable below is the convention
         // we are going to follow
```

Comment:
nit: Making the comment lines short doesn't help readability, as Xcode takes care of line wrapping so we have no practical limits, and less code fits on the screen (we all have screens wider than they are tall). Some manual wrapping is fine, up to you ultimately as I don't want us to be too prescriptive on that topic.
`// For LP variable below is the convention we are going to follow`. reads fine.